### PR TITLE
Add dex build pipeline (#566)

### PR DESCRIPTION
**Pull request by serglupach**

### CODE REVIEW

File: jenkins-pipelines/build-dex.Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+    environment {
+        build_workspace = "${WORKSPACE}"
+        def dex_dockerimage = "k8s-dex"
+        shared_lib_path = "deploy/legionPipeline.groovy"
+        //parameters section
+        //git branch to checkout 
+        //Set main branch for Job SCM GitBranch
+        //git branch to checkout legion
+        git_branch_legion = "${params.GitBranchLegion}"
```
Comment (alexander-semenets) [inline jenkins-pipelines/build-dex.Jenkinsfile]: can we use camelCase variables naming which is common groovy variables naming convention

File: jenkins-pipelines/build-dex.Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+pipeline {
+    agent any
+    options{
+        buildDiscarder(logRotator(numToKeepStr: '5'))
+        disableConcurrentBuilds()
+    }
+    environment {
+        build_workspace = "${WORKSPACE}"
+        def dex_dockerimage = "k8s-dex"
```
Comment (alexander-semenets) [inline jenkins-pipelines/build-dex.Jenkinsfile]: why do we need 'def' for dex_dockerimge?

**Review by alexander-semenets — CHANGES_REQUESTED** [review]

File: jenkins-pipelines/build-dex.Jenkinsfile (hunk 2 above)
Comment (serglupach) [reply jenkins-pipelines/build-dex.Jenkinsfile]: In this section not needed. Removed.

File: jenkins-pipelines/build-dex.Jenkinsfile (hunk 1 above)
Comment (serglupach) [reply jenkins-pipelines/build-dex.Jenkinsfile]: Update variables names in pipeline

**Review by alexander-semenets — DISMISSED** [review]

**Review by dsuslov — APPROVED** [review]
👍

**Pull request merged**